testing/fpgad/test_evt_c.cpp

```diff
@@ -0,0 +1,174 @@
+    ASSERT_TRUE(test_platform::exists(platform_key));
+    platform_ = test_platform::get(platform_key);
+    system_ = test_system::instance();
+    system_->initialize();
+    system_->prepare_syfs(platform_);
+
+    open_log(tmpfpgad_log_.c_str());
+    int i;
+    for (i = 0 ; i < MAX_SOCKETS ; ++i)
+      sem_init(&ap6_sem[i], 0, 0);
+  }
+
+  bool do_poll(int fd)
+  {
+    struct pollfd pollfd;
+
+    pollfd.fd = fd;
+    pollfd.events = POLL_IN;
+
+    int res = poll(&pollfd, 1, 1000);
+
+    return (res == 1) && ((pollfd.revents & POLLIN) != 0);
+  }
+
+  virtual void TearDown() override {
+    unregister_all_events();
+
+    int i;
+    for (i = 0 ; i < MAX_SOCKETS ; ++i)
+      sem_destroy(&ap6_sem[i]);
+
+    close_log();
+    system_->finalize();
+  }
+
+  std::string tmpfpgad_log_;
+  test_platform platform_;
+  test_system *system_;
+};
+
+/**
+ * @test       notify_err
+ * @brief      Test: evt_notify_error
+ * @details    evt_notify_error signals the event fd for each<br>
+ *             FPGA_EVENT_ERROR registration in event_registry_list.<br>
+ */
+TEST_P(fpgad_evt_c_p, notify_err) {
+  int conn_sockets[3] = { 0, 1, 2 };
+  int evt_fds[3]      = { -1, -1, eventfd(0, 0) };
+  const char *devs[3] = { "deva", "devb", "devc" }; 
+
+  EXPECT_NE(nullptr, register_event(conn_sockets[2], evt_fds[2], FPGA_EVENT_ERROR, devs[2]));
+  EXPECT_NE(nullptr, register_event(conn_sockets[1], evt_fds[1], FPGA_EVENT_ERROR, devs[1]));
+  EXPECT_NE(nullptr, register_event(conn_sockets[0], evt_fds[0], FPGA_EVENT_ERROR, devs[0]));
+
+  struct fpga_err err = {
+    .socket = -1,
+    .sysfsfile = "devc",
+    .reg_field = "reg_field",
+    .lowbit = 0,
+    .highbit = 1,
+    .occurred = true,
+    .callback = nullptr
+  };
+
+  evt_notify_error(&err);
+
+  ASSERT_TRUE(do_poll(evt_fds[2]));
+
+  close(evt_fds[2]);
+}
+
+/**
+ * @test       notify_ap6
+ * @brief      Test: evt_notify_ap6_and_null
+ * @details    evt_notify_ap6_and_null signals the event fd for each<br>
+ *             FPGA_EVENT_POWER_THERMAL registration in event_registry_list.<br>
+ */
+TEST_P(fpgad_evt_c_p, notify_ap6) {
+  int conn_sockets[3] = { 0, 1, 2 };
+  int evt_fds[3]      = { -1, eventfd(0, 0), -1 };
+  const char *devs[3] = { "deva", "devb", "devc" }; 
+
+  EXPECT_NE(nullptr, register_event(conn_sockets[2], evt_fds[2], FPGA_EVENT_POWER_THERMAL, devs[2]));
+  EXPECT_NE(nullptr, register_event(conn_sockets[1], evt_fds[1], FPGA_EVENT_POWER_THERMAL, devs[1]));
+  EXPECT_NE(nullptr, register_event(conn_sockets[0], evt_fds[0], FPGA_EVENT_POWER_THERMAL, devs[0]));
+
+  struct fpga_err err = {
+    .socket = 0,
+    .sysfsfile = "devb",
+    .reg_field = "reg_field",
+    .lowbit = 0,
+    .highbit = 1,
+    .occurred = true,
+    .callback = nullptr
+  };
+
+  evt_notify_ap6_and_null(&err);
+
+  ASSERT_TRUE(do_poll(evt_fds[1]));
+
+  close(evt_fds[1]);
+}
+
+INSTANTIATE_TEST_CASE_P(fpgad_evt_c, fpgad_evt_c_p,
+                        ::testing::ValuesIn(test_platform::keys()));
```

Comment:
This assumes that fpgad would work on other platforms, like PAC. I'm not sure if it's compatible yet...

Reply:
How would I make it specific to MCP?  Do a lookup on skx-p?